File: paper_experiments/code/R_algs/bic_score.R
#!/usr/bin/env Rscript
suppressMessages(library(pcalg))
suppressMessages(library(RcppCNPy))

args = commandArgs(trailingOnly=TRUE)
samples_filename = args[1]
amat_filename = args[2]
samples = npyLoad(samples_filename)
amat = npyLoad(amat_filename)
g = as(amat, "GaussParDAG")

score = new("GaussL0penObsScore", samples, use.cpp=FALSE)
s = score$global.score(g)
npySave('tmp_output.npy', s)

File: paper_experiments/code/R_algs/bic_score.py
import numpy as np
import os
import subprocess


def bic_score_R(dag, samples):
    amat, _ = dag.to_amat()
    samples_filename = 'tmp_samples.npy'
    np.save(samples_filename, samples)
    amat_filename = 'tmp_amat.npy'
    np.save(amat_filename, amat.astype(float))

    subprocess.call(['Rscript', 'paper_experiments/code/R_algs/bic_score.R', samples_filename, amat_filename])
    output_filename = 'tmp_output.npy'
    score = np.load(output_filename)

    os.remove(samples_filename)
    os.remove(amat_filename)
    os.remove(output_filename)
    return score


def bic_score_bnlearn_R(dag, samples):
    amat, o = dag.to_amat()
    samples_filename = 'tmp_samples.npy'
    np.save(samples_filename, samples)
    amat_filename = 'tmp_amat.npy'
    np.save(amat_filename, amat.astype(float))

    subprocess.call(['Rscript', 'paper_experiments/code/R_algs/bic_score_bnlearn.R', samples_filename, amat_filename])
    output_filename = 'tmp_output.npy'
    score = np.load(output_filename)

    os.remove(samples_filename)
    os.remove(amat_filename)
    os.remove(output_filename)
    return score
